### ISSUE TITLE
`command:<name>` event doesn't bubble to the parent

### ISSUE THREAD

**Issue by grabbou** [tj/commander.js#1402]
Hey,

Not sure if I am using this correctly, but it does seem to be a good use case and not far away from working. 

My program is as follows:
```
const config = program.commad('config');
const start = program.command('start');

program.on('command:*', () => /* handle unexpected command */);

program.on('command:config', () => /* never gets called */);
```

In the above example, when `config` command is executed, the `command:config` listener never gets executed. 

After console logging around the Commander source code, I have found out that the reason this doesn't work is that listener is registered on `parent` (in this context, main program) whereas the `emit` function is called on a subcommand (`config`). 

Changing `this.emit` to `this.parent.emit` fixes it https://github.com/tj/commander.js/blob/master/index.js#L1003

Is there any way to bubble events up? It seems to be quite natural to listen to `command:<name>` on the main object, otherwise, there's no difference from attaching a listener to `config` vs tweaking action handler.

**Comment by shadowspawn** [comment]
In your example, the emit for `command:config` should indeed be called on `program`. After some investigation, I think this is a bug introduced in Commander v5. The refactor of the code changed where the emit is called.

**Comment by shadowspawn** [comment]
Fix included in Commander 7.0.0.
